VAF Cutoff Filter Experiment for SNV callers

cansavvy opened the pull request:
### Purpose/implementation

This is a side analysis that originated from [this original PR and analysis](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/159#issuecomment-545096465). 

After doing some  initial comparisons of the callers TMB stats and finding them not well correlated to each other at all; the question from @jashapiro arose of whether increasing the VAF filter cutoff would help make the TMBs be more consistent across samples. Hence this analysis was born. 

### Issue

For SNV caller comparison #161, #103 and Tumor Mutation Burden #3 and sort of #11

### Directions for reviewers

See the rough draft of this notebook output from this notebook here: https://cansavvy.github.io/openpbta-notebook-concept/snv-callers/vaf_cutoff_experiment.nb.html

**Some questions for this PR:** 
What do you think of the current analyses? 
Are there other analyses you would like to see added? 

### Results

See the rough draft of this notebook output from this notebook here: https://cansavvy.github.io/openpbta-notebook-concept/snv-callers/vaf_cutoff_experiment.nb.html

**Preliminary conclusion:**
It looks like increasing the VAF cutoff improves the callers' TMB calculations in how much they are related to each other. However, note that although VarDicts TMBs get recovered a bit by increasing a VAF filter cutoff, VarDict's TMBs are still the least associated with the other callers'. 


#### Docker and continuous integration

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
Main _RPackages added to the Dockerfile:_
- UpSetR
- GGally 
- ggupset
- [x] This analysis has *NOT* been added to continuous integration.
Because there are no downstream results of this that will be used and it is a bit of a side question, I have not added this to the circleCI test. 

* [x] Run a linter
* [x] Set the seed- I didn't, but let me know if I should.
* [x] Comments and/or documentation up to date 
* [x] Double check your paths
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

cbethell commented on analyses/snv-callers/vaf_cutoff_experiment.Rmd:
````diff
@@ -0,0 +1,284 @@
+Function for creating a TMB correlation matrix amongst the callers for a particular cutoff.
+
+```{r}
+cor_matrix_plot <- function(cutoff) {
````
I like the idea of putting these custom functions in their own script in a `util` directory.

cansavvy replied:
@jashapiro What do you think about this idea? I'm kinda leaning toward it.

cbethell commented on analyses/snv-callers/vaf_cutoff_experiment.Rmd:
```diff
@@ -0,0 +1,284 @@
+filter? 
+
+### Summary of Findings:
+
```
Your preliminary conclusion seems like it would fit well here. Unless you were waiting for more conclusive results before updating this section.

cansavvy replied:
I'll add some more explanation of the findings here.

cbethell commented on analyses/snv-callers/vaf_cutoff_experiment.Rmd:
````diff
@@ -0,0 +1,284 @@
+## Make the TMB correlation plots
+
+```{r}
+lapply(unique(vaf_filter), cor_matrix_plot)
````
Out of curiosity, would further increasing the VAF cutoff continue to give a better correlation or is there a cutoff that you found to be optimal based on this analysis?

cansavvy replied:
After a certain cutoff, there are not many mutations left, but this range of VAF filter cutoffs seemed to give us a good enough idea of what was going on.

jashapiro commented on analyses/snv-callers/vaf_cutoff_experiment.Rmd:
````diff
@@ -0,0 +1,285 @@
+}
+```
+
+Calculate variance for a sample for a given caller amongst the different VAF cutoffs.
````
I don't understand what this is calculating. Variance of what? (Okay, I figured it out: Variance of TMB, but you should state that here. Also, I tend to prefer sd, as it tends to fit intuitions better.

jashapiro commented on analyses/snv-callers/vaf_cutoff_experiment.Rmd:
````diff
@@ -0,0 +1,285 @@
+lapply(unique(vaf_filter), cor_matrix_plot)
+```
+
+Get the variance for each sample across this experiment.
````
Again, explain here what you are calculating.

jashapiro commented on analyses/snv-callers/vaf_cutoff_experiment.Rmd:
```diff
@@ -0,0 +1,285 @@
+  ggplot2::geom_jitter(ggplot2::aes(color = sample)) +
+  ggplot2::theme_classic() +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Sample Variance Across VAF Filter Cutoffs") +
```
And label it here (Variance in TMB)

cansavvy commented:
> It isn't really a sampling question (you chose the cutoffs; they aren't random sets), so the changes in TMB aren't really something that variance makes sense for. I'd probably just eliminate that section.

The main point I was searching for here was "how much are individual sample's TMB's changing as we change cutoffs?" I get that variance is not quite appropriate for this, but wasn't sure how to show this. That being said, I am fine with dropping that whole section and moving on.

cansavvy commented:
Okay, @cbethell and @jashapiro, I've removed the confusing sample TMB variance section, added a summary at the top of the file, but did not make a separate `util` file (I've decided I prefer to keep the functions of this notebook as separate from the other more main-line items in this analysis). How do we feel about this PR now?
